go/obscuronode/enclave/state.go

```diff
@@ -84,7 +84,7 @@ func updateState(b *types.Block, blockResolver db.BlockResolver, storage db.Stor
 		return val
 	}
 
-	if blockResolver.HeightBlock(b) == 0 {
+	if b.NumberU64() == obscurocommon.L2GenesisHeight {
```

Comment:
Good spot. Had a better look at this check and it's not necessary at all, since there is another check below which makes sure the function does not do anything beyond the block where the genesis rollup is